VoodooPS2Trackpad/VoodooPS2SynapticsTouchPad.cpp

```diff
@@ -2621,10 +2621,16 @@ void ApplePS2SynapticsTouchPad::unregisterHIDPointerNotifications()
 void ApplePS2SynapticsTouchPad::notificationHIDAttachedHandlerGated(IOService * newService,
                                                                     IONotifier * notifier)
 {
-    char path[256];
-    int len = 255;
-    memset(path, 0, len);
-    newService->getPath(path, &len, gIOServicePlane);
+    int len = 256;
+    char *path = IONew(char, len);
```

Comment:
1. This doesn't improve anything: there is no reason to allocate a temporary array in the heap. 256 bytes is not very large.
2. This even makes it worse: `path` is not freed, so there is a memory leak. Also, as `path` is only used for log messages, old implementation worked even if the buffer size was insufficient, but the new one works only if the buffer size is sufficient, else it just skips the notification. 

Either revert it, or convert it into actual improvement by reallocating the buffer if `getPath` returned `false` and specified the required length in `len` inout parameter and freeing it at all returns.